src/lid.c

```diff
@@ -181,14 +181,17 @@ void lid_init(void) {
 		destroy_libinput_discovery(libinput_discovery);
 
 		if (!device_path) {
+			log_warn("No lid device was discovered; lid remains unitialized");
```

Comment:
Please remove this line; it was annoying for people who were not using laptops.

Reply:
Will do.

I did check my logs though, and this is exactly the code path was leading to the segfault I got:

    May 30 12:33:43 laptop systemd[1043]: Started way-displays.service.
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43] way-displays version 1.5.3-SNAPSHOT
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43] Found configuration file: /home/joost/src/dot/sway/way-displays/cfg.yaml
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43] Active configuration:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   Arrange in a ROW aligned at the TOP
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   Order:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     Sharp Corporation 0x149A
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     Unknown 0xA029
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     DELL U2718Q FN84K92F0C1L
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     DELL U2715H GH85D83B002S
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     DELL U2715H GH85D83B05RS
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   Auto scale: OFF
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   Scale:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     Sharp Corporation 0x149A: 1.000
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     Unknown 0xA029: 2.000
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     DELL U2718Q FN84K92F0C1L: 1.000
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     DELL U2715H GH85D83B002S: 1.000
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     DELL U2715H GH85D83B05RS: 1.000
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event13 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event14 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event1 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event0 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event2 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event11 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event10 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event12 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event3 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event4 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event15 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event16 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event17 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event18 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event19 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event20 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event7 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event6 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event9 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event5 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: libinput open /dev/input/event8 failed: 13 Permission denied
    May 30 12:33:43 laptop way-displays[24027]: W [12:33:43] WARNING: No lid device was discovered; lid remains unitialized
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43] DP-6 Arrived:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   info:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     name:     'DP-6'
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     desc:     'Dell Inc. DELL U2715H GH85D83B05RS (DP-6)'
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     width:    600mm
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     height:   340mm
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     dpi:      107.97 @ 2560x1440
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     2560 x 1440 @  60 Hz   59,951 mHz (preferred)
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     2048 x 1152 @  60 Hz   60,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1200 @  60 Hz   59,950 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  60 Hz   60,000 mHz  60,000 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  50 Hz   50,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  30 Hz   30,000 mHz  29,970 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  25 Hz   25,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  24 Hz   24,000 mHz  23,976 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1680 x 1050 @  60 Hz   59,883 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1600 x 1200 @  60 Hz   60,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x 1024 @  75 Hz   75,025 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x 1024 @  60 Hz   60,020 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x  720 @  60 Hz   60,000 mHz  60,000 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x  720 @  50 Hz   50,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1152 x  864 @  75 Hz   75,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1024 x  768 @  75 Hz   75,029 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1024 x  768 @  60 Hz   60,004 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      800 x  600 @  75 Hz   75,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      800 x  600 @  60 Hz   60,317 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      720 x  576 @  50 Hz   50,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      720 x  480 @  60 Hz   60,000 mHz  60,000 mHz  59,940 mHz  59,940 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      720 x  400 @  70 Hz   70,082 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      640 x  480 @  75 Hz   75,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      640 x  480 @  60 Hz   60,000 mHz  59,940 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   current:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     scale:    1.000
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     position: 4480,0
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     2560x1440@60Hz (59,951mHz) (preferred)
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43] DP-4 Arrived:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   info:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     name:     'DP-4'
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     desc:     'Dell Inc. DELL U2715H GH85D83B002S (DP-4)'
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     width:    600mm
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     height:   340mm
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     dpi:      107.97 @ 2560x1440
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     2560 x 1440 @  60 Hz   59,951 mHz (preferred)
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     2048 x 1152 @  60 Hz   60,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1200 @  60 Hz   59,885 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  60 Hz   60,000 mHz  60,000 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  50 Hz   50,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  30 Hz   30,000 mHz  29,970 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  25 Hz   25,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1920 x 1080 @  24 Hz   24,000 mHz  23,976 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1680 x 1050 @  60 Hz   59,954 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1600 x 1200 @  60 Hz   60,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x 1024 @  75 Hz   75,025 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x 1024 @  60 Hz   60,020 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x  720 @  60 Hz   60,000 mHz  60,000 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1280 x  720 @  50 Hz   50,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1152 x  864 @  75 Hz   75,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1024 x  768 @  75 Hz   75,029 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     1024 x  768 @  60 Hz   60,004 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      800 x  600 @  75 Hz   75,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      800 x  600 @  60 Hz   60,317 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      720 x  576 @  50 Hz   50,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      720 x  480 @  60 Hz   60,000 mHz  60,000 mHz  59,940 mHz  59,940 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      720 x  400 @  70 Hz   70,082 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      640 x  480 @  75 Hz   75,000 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:      640 x  480 @  60 Hz   60,000 mHz  59,940 mHz  59,940 mHz
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   current:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     scale:    1.000
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     position: 1920,0
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     2560x1440@60Hz (59,951mHz) (preferred)
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43] eDP-1 Arrived:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   info:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     name:     'eDP-1'
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     desc:     'Unknown 0xA029 0x00000000 (eDP-1)'
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     width:    340mm
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     height:   190mm
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     dpi:      287.81 @ 3840x2160
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     3840 x 2160 @  60 Hz   60,000 mHz (preferred)
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]   current:
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     scale:    2.000
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     position: 0,0
    May 30 12:33:43 laptop way-displays[24027]: I [12:33:43]     mode:     3840x2160@60Hz (60,000mHz) (preferred)
    May 30 12:33:43 laptop systemd-coredump[24033]: Process 24027 (way-displays) of user 1000 dumped core.
                                                        
                                                        Module /home/joost/src/way-displays/way-displays with build-id 022712a782117c56df55f6a92296fd3a1bced2c1
                                                        Module linux-vdso.so.1 with build-id 91a0b49aeeb7320c76108fc5fdf013b225461390
                                                        Module libpthread.so.0 with build-id 95ae4f30a6f12ccbff645d30f8e1a3ee23ec7d36
                                                        Module libpcre.so.1 with build-id 845483dd0acba86de9f0313102bebbaf3ce52767
                                                        Module libglib-2.0.so.0 with build-id e268e80e668ce871312cf50e9969f88b086e33f2
                                                        Module libgobject-2.0.so.0 with build-id 4337ec84b2f33ec39a95863d47bfca0a28afd488
                                                        Module libgudev-1.0.so.0 with build-id 460465b63a086d945124662363191903c0002793
                                                        Module ld-linux-x86-64.so.2 with build-id fc93487393eea02b5bc6e76e48976fc325294c24
                                                        Module libwacom.so.9 with build-id dab3ab00bae4a88ba3bd0d2e9262a9d5bf59187b
                                                        Module libevdev.so.2 with build-id 09eb82a0dfca98325f54b3918875cff09a5c1428
                                                        Module libmtdev.so.1 with build-id 862cf8d5b0a0dd4d4d225e17f54be93ab9c3b34e
                                                        Module libffi.so.8 with build-id f0a9586cf0f42d2b9971bd1065ca3a6b19f4a2c2
                                                        Module libc.so.6 with build-id 388993b6ef62f964bc7bf473c069fbfe957b9e44
                                                        Module libgcc_s.so.1 with build-id 0e3de903950e35ae59a5de8c00b1817a4a71ca01
                                                        Module libm.so.6 with build-id 210ec9905e41825671210f8f7d0b24d6c371196a
                                                        Module libstdc++.so.6 with build-id b3eaebcb5369487562e52dd8d016c9cd77f80c54
                                                        Module libudev.so.1 with build-id ac4cd7595ed265c7be79d2aedf407ca43e17a890
                                                        Module libinput.so.10 with build-id 2dae1736f9a02020c3932f0ceb476474463a5c07
                                                        Module libyaml-cpp.so.0.7 with build-id df7f91eca5b07ab1042be32f1b9a7c4e6fa88230
                                                        Module libwayland-client.so.0 with build-id 95e7368b400dd57e3db2a5c385de71c7dca08879
                                                        Stack trace of thread 24027:
                                                        #0  0x000055cff78c4633 n/a (/home/joost/src/way-displays/way-displays + 0xd633)
                                                        ELF object binary architecture: AMD x86-64
    May 30 12:33:43 laptop systemd[1043]: way-displays.service: Main process exited, code=dumped, status=11/SEGV
    May 30 12:33:43 laptop systemd[1043]: way-displays.service: Failed with result 'core-dump'.

Seeing that reaching this branch in the code could be an indication that the system has a misconfiguration, it could be useful to leave the message in, but to reduce the severity of the log message to INFO or DEBUG, something like `"No lid was discovered on device %s"`.